stellargraph/layer/gcn_lstm.py

```diff
@@ -0,0 +1,349 @@
+
+        Args:
+            inputs (ndarray): node features (size B x T x N),
+                where B is the batch size,
+                      T is the sequence length, and
+                      N is the number of nodes in the graph.
+
+        Returns:
+            Keras Tensor that represents the output of the layer.
+        """
+
+        # Calculate the layer operation of GCN
+
+        h_graph = K.dot(features, self.A)
+        output = tf.transpose(
+            K.dot(tf.transpose(h_graph, [0, 2, 1]), self.kernel), [0, 2, 1]
+        )
+
+        # Add optional bias & apply activation
+        if self.bias is not None:
+            output += self.bias
+
+        output = self.activation(output)
+
+        return output
+
+
+@experimental(reason="Lack of unit tests and some code refinement", issues=[1131, 1132])
+class GraphConvolutionLSTM:
+
+    """
+        A stack of 2 Graph Convolutional layers followed by an LSTM, Dropout and,  Dense layer.
+        This main components of GNN architecture is inspired by: T-GCN: A Temporal Graph Convolutional Network for Traffic Prediction
+                                          (https://arxiv.org/abs/1811.05320)
+        The implementation of the above paper is based on one graph convolution layer stacked with a GRU layer.
+        The StellarGraph implementation is built as a stack of the following layers:
+           1. 2-layer graph convolutional network (gcn) proposed by Kipf & Welling in  https://arxiv.org/abs/1609.02907 (ICLR 2017).
+           2. 1 LSTM layer
+           3. 1 Dense layer
+           4. 1 Dropout layer
+           The last two layers consistently showed better performance and regularization experimentally.
+        Notes:
+          - The normalized Lapalacian matrix is provided as input to
+            Keras methods. 
+            
+        Args:
+            layer_sizes (list of int): Output sizes of GCN layers in the stack.
+            bias (bool): If True, a bias vector is learnt for each layer in the GCN model.
+            dropout (float): Dropout rate applied to input features of each GCN layer.
+            activations (list of str or func): Activations applied to each layer's output;
+                defaults to ['relu', ..., 'relu'].
+            kernel_initializer (str or func, optional): The initialiser to use for the weights of each layer.
+            kernel_regularizer (str or func, optional): The regulariser to use for the weights of each layer.
+            kernel_constraint (str or func, optional): The constraint to use for the weights of each layer.
+            bias_initializer (str or func, optional): The initialiser to use for the bias of each layer.
+            bias_regularizer (str or func, optional): The regulariser to use for the bias of each layer.
+            bias_constraint (str or func, optional): The constraint to use for the bias of each layer.
+        """
+
+    def __init__(
+        self,
+        outputs,
```

Comment:
I think this argument is number of nodes? we can just derive this from `adj` so I think we can leave this out
```suggestion
```